tools/isobuild/compiler-plugin.js

```diff
@@ -1394,8 +1397,19 @@ export class PackageSourceBatch {
     if (canCache) {
       LINKER_CACHE.set(cacheKey, ret);
       if (cacheFilename) {
+        const cacheKeySuffixRegex = new RegExp('_' + cacheKeySuffix, 'g');
+        const oldCacheFilename =
+          cacheFilename.replace(cacheKeySuffixRegex, '_*');
         // Write asynchronously.
-        Fiber(() => files.writeFileAtomically(cacheFilename, retAsJSON)).run();
+        Fiber(() => {
+          try {
+            files.rm_recursive(oldCacheFilename);
```

Comment:
Would it make sense to synchronously get the list of filenames in the folder and then start the remove and write process in parallel, where the remove process would use `folder/file.ext` from the list?